### ISSUE TITLE
Winget_Source: Deployment failed #0: 0x8000FFFF 

### ISSUE THREAD

**Issue by richardmalone** [microsoft/winget-cli#276]
Work laptop, latest Windows 10 release, running as Administrator but there are group policy restrictions in place. 

Failed to install winget using DesktopAppInstaller. CLI tools worked but took additional and unexpected steps.  

Double clicking the AppXBundle file gave a deployment error 0x8000FFFF, but using the  Add-AppxPackage worked fine. However, running `winget source update` gave an additional error (log below). I resolved this by downloading the sources from https://winget.azureedge.net/cache/source.msix and installing through Add-AppxPackage.

Winget now appears to be working correctly. 

Logfile output is:

> 2020-5-21 21:24:35.580 [CORE] WinGet, version [0.1.41331], activity [{1757EE48-D4BB-4BB5-8B7B-D39CE709534F}]
> 2020-5-21 21:24:35.581 [CORE] OS: Windows.Desktop v10.0.18363.836
> 2020-5-21 21:24:35.581 [CORE] Package: Microsoft.DesktopAppInstaller v1.0.41331.0
> 2020-5-21 21:24:35.581 [CLI ] WinGet invoked with arguments: 'source' 'update'
> 2020-5-21 21:24:35.581 [CLI ] Found subcommand: source
> 2020-5-21 21:24:35.581 [CLI ] Found subcommand: update
> 2020-5-21 21:24:35.581 [CLI ] Leaf command to execute: root:source:update
> 2020-5-21 21:24:35.598 [CLI ] Executing command: update
> 2020-5-21 21:24:35.605 [REPO] Named source to be updated, found: winget
> 2020-5-21 21:24:35.605 [REPO] Initializing source from: winget => https://winget.azureedge.net/cache/source.msix
> 2020-5-21 21:24:36.996 [REPO] Found package full name: winget => Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe
> 2020-5-21 21:24:37.026 [CORE] Examining extension: PFN = Microsoft.Winget.Source_8wekyb3d8bbwe, ID = IndexDB
> 2020-5-21 21:24:37.026 [CORE] Found matching extension.
> 2020-5-21 21:24:37.636 [CORE] Downloading to path: "C:\\Users\\rmalone\\AppData\\Local\\Packages\\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\\TempState\\WinGet\\Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix"
> 2020-5-21 21:24:37.637 [CORE] Started applying motw to "C:\\Users\\rmalone\\AppData\\Local\\Packages\\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\\TempState\\WinGet\\Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix"
> 2020-5-21 21:24:37.637 [CORE] Finished applying motw
> 2020-5-21 21:24:37.637 [CORE] Downloading from url: https://winget.azureedge.net/cache/source.msix
> 2020-5-21 21:24:38.224 [CORE] Download completed.
> 2020-5-21 21:24:38.225 [CORE] Starting RequestAddPackage operation #0: file:///C:/Users/rmalone/AppData/Local/Packages/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe/TempState/WinGet/Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix
> 2020-5-21 21:24:38.227 [CORE] Begin waiting for deployment #0
> 2020-5-21 21:24:38.227 [CORE] Begin blocking for deployment #0
> 2020-5-21 21:24:38.610 [CORE] Deployment failed #0: Deployment Add operation with target volume C: on Package Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe from:  (Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix)  failed with error 0x8000FFFF. See http://go.microsoft.com/fwlink/?LinkId=235160 for help diagnosing app deployment issues.
> 2020-5-21 21:24:38.610 [FAIL] C:\BA\138\s\external\pkg\src\AppInstallerCommonCore\Deployment.cpp(50)\AppInstallerCLI.exe!00007FF69CE4ED54: (caller: 00007FF69CE4F3B4) Exception(1) tid(a34) 8000FFFF Catastrophic failure
> 
>     Msg:[Install failed: Deployment Add operation with target volume C: on Package Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe from:  (Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix)  failed with error 0x8000FFFF. See http://go.microsoft.com/fwlink/?LinkId=235160 for help diagnosing app deployment issues.] 
> 
> 2020-5-21 21:24:38.611 [FAIL] C:\BA\138\s\external\pkg\src\AppInstallerRepositoryCore\RepositorySource.cpp(229)\AppInstallerCLI.exe!00007FF69CEBFB31: (caller: 00007FF69CE70AAC) LogHr(1) tid(a34) 8000FFFF Catastrophic failure
> 
>     Msg:[C:\BA\138\s\external\pkg\src\AppInstallerCommonCore\Deployment.cpp(50)\AppInstallerCLI.exe!00007FF69CE4ED54: (caller: 00007FF69CE4F3B4) Exception(1) tid(a34) 8000FFFF Catastrophic failure
> 
>     Msg:[Install failed: Deployment Add operation with target volume C: on Package Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe from:  (Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix)  failed with error 0x8000FFFF. See http://go.microsoft.com/fwlink/?LinkId=235160 for help diagnosing app deployment issues.] 
> ] 
> 
> 2020-5-21 21:24:38.611 [REPO] Source update failed, waiting a bit and retrying: winget
> 2020-5-21 21:24:40.622 [CORE] Examining extension: PFN = Microsoft.Winget.Source_8wekyb3d8bbwe, ID = IndexDB
> 2020-5-21 21:24:40.622 [CORE] Found matching extension.
> 2020-5-21 21:24:41.277 [CORE] Downloading to path: "C:\\Users\\rmalone\\AppData\\Local\\Packages\\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\\TempState\\WinGet\\Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix"
> 2020-5-21 21:24:41.278 [CORE] Started applying motw to "C:\\Users\\rmalone\\AppData\\Local\\Packages\\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\\TempState\\WinGet\\Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix"
> 2020-5-21 21:24:41.278 [CORE] Finished applying motw
> 2020-5-21 21:24:41.279 [CORE] Downloading from url: https://winget.azureedge.net/cache/source.msix
> 2020-5-21 21:24:41.482 [CORE] Download completed.
> 2020-5-21 21:24:41.482 [CORE] Starting RequestAddPackage operation #1: file:///C:/Users/rmalone/AppData/Local/Packages/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe/TempState/WinGet/Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix
> 2020-5-21 21:24:41.484 [CORE] Begin waiting for deployment #1
> 2020-5-21 21:24:41.484 [CORE] Begin blocking for deployment #1
> 2020-5-21 21:24:41.872 [CORE] Deployment failed #1: Deployment Add operation with target volume C: on Package Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe from:  (Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix)  failed with error 0x8000FFFF. See http://go.microsoft.com/fwlink/?LinkId=235160 for help diagnosing app deployment issues.
> 2020-5-21 21:24:41.872 [FAIL] C:\BA\138\s\external\pkg\src\AppInstallerCommonCore\Deployment.cpp(50)\AppInstallerCLI.exe!00007FF69CE4ED54: (caller: 00007FF69CE4F3B4) Exception(2) tid(a34) 8000FFFF Catastrophic failure
> 
>     Msg:[Install failed: Deployment Add operation with target volume C: on Package Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe from:  (Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix)  failed with error 0x8000FFFF. See http://go.microsoft.com/fwlink/?LinkId=235160 for help diagnosing app deployment issues.] 
> 
> 2020-5-21 21:24:41.872 [CLI ] Caught wil::ResultException: C:\BA\138\s\external\pkg\src\AppInstallerCommonCore\Deployment.cpp(50)\AppInstallerCLI.exe!00007FF69CE4ED54: (caller: 00007FF69CE4F3B4) Exception(2) tid(a34) 8000FFFF Catastrophic failure
> 
>     Msg:[Install failed: Deployment Add operation with target volume C: on Package Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe from:  (Microsoft.Winget.Source_2020.521.2050.685_neutral__8wekyb3d8bbwe.msix)  failed with error 0x8000FFFF. See http://go.microsoft.com/fwlink/?LinkId=235160 for help diagnosing app deployment issues.] 
> 
> 

Environment is:

> Windows Package Manager v0.1.41331 Preview
> Copyright (c) Microsoft Corporation. All rights reserved.
> 
> Windows: Windows.Desktop v10.0.18363.836
> Package: Microsoft.DesktopAppInstaller v1.0.41331.0

**Comment by Swellenator** [comment]
I had this problem also, thank you for the workaround steps.

**Comment by sadunsevingen** [comment]
I've the same problem when I applied same steps above it's fixed.

**Comment by ibigpapa** [comment]
the work around did work.  unfortunately after installing a few packages i'm starting to get this error on all packages, even running `winget source update`

**Comment by rkasbohm** [comment]
We are having the same issue.  For us, it appears to be company-wide.  We don't use AD or GP.  We do use Workspace One.  We think it might have to do with that.

**Comment by aheidelberg** [comment]
I've had a similar issue but your workaround worked for me as well.

`Failed to open the source; try removing and re-adding it
An unexpected error occurred while executing the command:
Install failed: Deployment Add operation with target volume C: on Package Microsoft.Winget.Source_2020.718.1720.43_neutral__8wekyb3d8bbwe from:  (Microsoft.Winget.Source_8wekyb3d8bbwe.msix)  failed with error 0x80073D1B. See http://go.microsoft.com/fwlink/?LinkId=235160 for help diagnosing app deployment issues.
0x80073d1b : The packages failed the SmartScreen reputation check.`

I'm using CoreDNS as a DNS Server, which forwards everything to `8.8.8.8` except requests to `*.test`, maybe this is an issue?

**Comment by RPitt** [comment]
Just here to say me too ... I've spent over an hour and got no further than the original poster.

The latest 0.2.2941 AppInstallerCLI.exe is still failing with `Exception(1) tid(48f0) 8000FFFF Catastrophic failure` in the logs.

**Comment by cryolithic** [comment]
I've been fighting this error for weeks now. It might work for a bit, then would fail to update sources, so I would try everything with no luck until the next time I attempted to use it and would succeed for unknown reasons until it stopped again.

Always resulting in some error similar to 
```
An unexpected error occurred while executing the command:
Install failed: Deployment Add operation with target volume C: on Package Microsoft.Winget.Source_2020.1211.1927.552_neutral__8wekyb3d8bbwe from:  (Microsoft.Winget.Source_8wekyb3d8bbwe.msix)  failed with error 0x8000FFFF. See http://go.microsoft.com/fwlink/?LinkId=235160 for help diagnosing app deployment issues.
0x8000ffff : Catastrophic failure

```
Today this command worked, though I'm not sure which force was the good one.
`Invoke-WebRequest -Uri https://winget.azureedge.net/cache/source.msix -OutFile source.msix`

```
Add-AppxPackage -Path .\source.msix -ForceUpdateFromAnyVersion -Confirm -ForceTargetApplicationShutdown

Confirm
Are you sure you want to perform this action?
Performing the operation "Deploy package" on target "C:\Users\jtrudel\source.msix".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): a
```

```
winget source update
Updating all sources...
Updating source: winget...
Done
Updating source: msstore...
Done
```

**Comment by rkasbohm** [comment]
Wow, that just seemed to fix my issue....will know after more testing.

Thanks @cryolithic ,

**Comment by deftdawg** [comment]
Thanks @cryolithic your steps helped... 

I was getting this:
```
C:\Windows\system32> winget search vlc
Failed to open the source; try the 'source reset' command if the problem persists.
An unexpected error occurred while executing the command:
Install failed: error 0x8000FFFF: Windows cannot create the repository metadata for the Microsoft.Winget.Source_2021.115.2440.659_neutral__8wekyb3d8bbwe package with 0 apps.
0x8000ffff : Catastrophic failure
```

These are the all steps for other windows noobs like me (I'm arriving from OSX):

- open powershell as *admin* (I tried to use cmd, however the instructions are PS specific)
```sh
cd $HOME\AppData\Local\Microsoft\WindowsApps
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12 # Fix "Invoke-WebRequest : The request was aborted: Could not create SSL/TLS secure channel."
Invoke-WebRequest -Uri https://winget.azureedge.net/cache/source.m
six -OutFile source.msix
Add-AppxPackage -Path .\source.msix -Confirm -ForceTargetApplicationShutdown # NOTE: "-ForceUpdateFromAnyVersion" was not a valid arg
winget source reset --force
winget search vlc # works
```

**Comment by denelon** [comment]
This Issue should be resolved with changes that were made a few releases ago related to the Mark of the Web.

**Comment by RPitt** [comment]
Hi @denelon are you sure about this?

FYI I just tried downloading/running the latest [v-0.2.10971-preview](https://github.com/microsoft/winget-cli/releases/tag/v-0.2.10971-preview) and it still fails for me with same

> App installation failed with error message: Deployment Add operation with target volume C: on Package Microsoft.DesktopAppInstaller_2021.407.1917.0_neutral_~_8wekyb3d8bbwe from:  (Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.appxbundle)  failed with error 0x8000FFFF.

Also tried 'unblocking' it to reset it's InternetZone, same result.

Whilst this could be something special about my machine, I suspect as others have reported it's a common problem.  
If anyone wants me to gather detailed logs or anything let me know.

**Comment by denelon** [comment]
@RPitt thank you for responding. Yes, please submit feedback via [Windows]+[f] and select "Apps / Windows Package Manager". That will send us logs to diagnose.

**Comment by denelon** [comment]
This looks like an issue with installing the "App Installer". We're going to look through logs to see if we have the information necessary to troubleshoot. This is being marked as "Area-External" since it's not related to the Windows Package Manger itself. It's really related to how the App Installer is handled in Windows deployment.

**Comment by JohnMcPMS** [comment]
I believe that the original issue was resolved with #627.  If there are issues with installing the DesktopAppInstaller package itself I think that a new issue should be opened with details from a more current release, but note that those would likely be external to winget-cli as that is Windows MSIX deployment related.  Filing feedback via Feedback Hub is the best way to get Windows issues on the right teams.
